quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/CachingPropertyMappers.java

```diff
@@ -152,4 +163,23 @@ private static PropertyMapper<?> maxCountOpt(String cacheName, BooleanSupplier i
               .paramLabel("max-count")
               .build();
     }
+
+    private static boolean isRemoteCacheHostEnabled() {
+        return InfinispanUtils.isRemoteInfinispan() || Profile.isFeatureEnabled(Profile.Feature.CACHE_EMBEDDED_REMOTE_STORE);
+    }
+
+    private static  <T> void validateIsPresent(PropertyMapper<T> mapper, ConfigValue value) {
```

Comment:
I don't think this will currently work - we only validate values the user supplies, not ones they don't. We can certainly make changes for this - it would be useful as well eventually for requiring the db to be set in the production profile.

Reply:
When I found out this didn't work, I fixed it by adding the `isRequired` to the `PropertyMapper`. I'm checking it in the `Picocli`.